[BUG]在与"无名氏插件"同时装载会无法生成假人

服务器控制台显示加载成功,但是聊天栏输入ffpp/假人创建不会有反应
(感觉是无法识别玩家消息)
[[链接-无名氏插件]](https://d.usfpack.site/public/index.php)

第一次使用github提交BUG不是懂很多😅
已执行但未生成build文件夹
```
~/.../aacc/ym $ npm i -g typescript   ↵ 

added 1 package in 1m

~/.../aacc/ym $ npm install    ↵ 

up to date in 2m

35 packages are looking for funding
  run 'npm fund' for details

~/.../aacc/ym $ npm run build   ↵ 

> build
> cd . && tsc && webpack && node packer.js

tscripts/xTerrain/main.ts:94:57 - error TS2322: Type 'import("/data/data/com.termux/files/home/Desktop/aacc/ym/node_modules/@minecraft/server/index").Dimension' is not assignable to type 'import("/data/data/com.termux/files/home/Desktop/aacc/ym/node_modules/@minecraft/server-gametest/node_modules/@minecraft/server/index").Dimension'.
  Types of property 'createExplosion' are incompatible.
    Type '(location: import("/data/data/com.termux/files/home/Desktop/aacc/ym/node_modules/@minecraft/server/index").Vector3, radius: number, explosionOptions?: import("/data/data/com.termux/files/home/Desktop/aacc/ym/node_modules/@minecraft/server/index").ExplosionOptions) => boolean' is not assignable to type '(location: import("/data/data/com.termux/files/home/Desktop/aacc/ym/node_modules/@minecraft/server-gametest/node_modules/@minecraft/server/index").Vector3, radius: number, explosionOptions?: import("/data/data/com.termux/files/home/Desktop/aacc/ym/node_modules/@minecraft/server-gametest/node_modules/@minecraft/serve...'.
      Types of parameters 'explosionOptions' and 'explosionOptions' are incompatible.
        Type 'import("/data/data/com.termux/files/home/Desktop/aacc/ym/node_modules/@minecraft/server-gametest/node_modules/@minecraft/server/index").ExplosionOptions' is not assignable to type 'import("/data/data/com.termux/files/home/Desktop/aacc/ym/node_modules/@minecraft/server/index").ExplosionOptions'.
          The types of 'source.dimension' are incompatible between these types.
            Type 'Dimension' is missing the following properties from type 'Dimension': findClosestBiome, getBlockAbove, getBlockBelow, getWeather

94             simulatedPlayer.setSpawnPoint({...location, dimension})
                                                           ~~~~~~~~~

  node_modules/@minecraft/server-gametest/node_modules/@minecraft/server/index.d.ts:15949:5
    15949     dimension: Dimension;
              ~~~~~~~~~
    The expected type comes from property 'dimension' which is declared here on type 'DimensionLocation'

tscripts/xTerrain/main.ts:95:49 - error TS2322: Type 'import("/data/data/com.termux/files/home/Desktop/aacc/ym/node_modules/@minecraft/server/index").Dimension' is not assignable to type 'import("/data/data/com.termux/files/home/Desktop/aacc/ym/node_modules/@minecraft/server-gametest/node_modules/@minecraft/server/index").Dimension'.
  Types of property 'createExplosion' are incompatible.
    Type '(location: import("/data/data/com.termux/files/home/Desktop/aacc/ym/node_modules/@minecraft/server/index").Vector3, radius: number, explosionOptions?: import("/data/data/com.termux/files/home/Desktop/aacc/ym/node_modules/@minecraft/server/index").ExplosionOptions) => boolean' is not assignable to type '(location: import("/data/data/com.termux/files/home/Desktop/aacc/ym/node_modules/@minecraft/server-gametest/node_modules/@minecraft/server/index").Vector3, radius: number, explosionOptions?: import("/data/data/com.termux/files/home/Desktop/aacc/ym/node_modules/@minecraft/server-gametest/node_modules/@minecraft/serve...'.
      Types of parameters 'explosionOptions' and 'explosionOptions' are incompatible.
        Type 'import("/data/data/com.termux/files/home/Desktop/aacc/ym/node_modules/@minecraft/server-gametest/node_modules/@minecraft/server/index").ExplosionOptions' is not assignable to type 'import("/data/data/com.termux/files/home/Desktop/aacc/ym/node_modules/@minecraft/server/index").ExplosionOptions'.
          The types of 'source.dimension' are incompatible between these types.
            Type 'Dimension' is missing the following properties from type 'Dimension': findClosestBiome, getBlockAbove, getBlockBelow, getWeather

95             simulatedPlayer.teleport(location, {dimension})
                                                   ~~~~~~~~~

  node_modules/@minecraft/server-gametest/node_modules/@minecraft/server/index.d.ts:17294:5
    17294     dimension?: Dimension;
              ~~~~~~~~~
    The expected type comes from property 'dimension' which is declared here on type 'TeleportOptions'

tscripts/xTerrain/plugins/breakBlock.ts:58:27 - error TS2352: Conversion of type 'import("/data/data/com.termux/files/home/Desktop/aacc/ym/node_modules/@minecraft/server-gametest/node_modules/@minecraft/server/index").Dimension' to type 'import("/data/data/com.termux/files/home/Desktop/aacc/ym/node_modules/@minecraft/server/index").Dimension' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
  Type 'Dimension' is missing the following properties from type 'Dimension': findClosestBiome, getBlockAbove, getBlockBelow, getWeather

58         const dimension = <Dimension>man.dimension
                             ~~~~~~~~~~~~~~~~~~~~~~~~

tscripts/xTerrain/plugins/youAreMine.ts:38:15 - error TS2352: Conversion of type 'EntityComponent' to type 'EntityEquippableComponent' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
  Type 'EntityComponent' is missing the following properties from type 'EntityEquippableComponent': getEquipment, getEquipmentSlot, setEquipment

38     const s = <EntityEquippableComponent>SimPlayer.getComponent("minecraft:equippable")
                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

tscripts/xTerrain/plugins/youAreMine.ts:56:15 - error TS2352: Conversion of type 'EntityComponent' to type 'EntityEquippableComponent' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
  Type 'EntityComponent' is missing the following properties from type 'EntityEquippableComponent': getEquipment, getEquipmentSlot, setEquipment

56     const s = <EntityEquippableComponent>SimPlayer.getComponent("minecraft:equippable")
                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

tscripts/xTerrain/plugins/youAreMine.ts:75:16 - error TS2352: Conversion of type 'EntityComponent' to type 'EntityInventoryComponent' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
  Type 'EntityComponent' is missing the following properties from type 'EntityInventoryComponent': additionalSlotsPerStrength, canBeSiphonedFrom, containerType, inventorySize, and 2 more.

75     const s = (<EntityInventoryComponent>SimPlayer.getComponent("minecraft:inventory")).container
                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

tscripts/xTerrain/plugins/youAreMine.ts:105:15 - error TS2352: Conversion of type 'EntityComponent' to type 'EntityEquippableComponent' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
  Type 'EntityComponent' is missing the following properties from type 'EntityEquippableComponent': getEquipment, getEquipmentSlot, setEquipment

105     const s = <EntityEquippableComponent>SimPlayer.getComponent("minecraft:equippable") // SimPlayer
                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

tscripts/xTerrain/plugins/youAreMine.ts:132:19 - error TS2352: Conversion of type 'EntityComponent' to type 'EntityEquippableComponent' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
  Type 'EntityComponent' is missing the following properties from type 'EntityEquippableComponent': getEquipment, getEquipmentSlot, setEquipment

132     const equip = <EntityEquippableComponent>SimPlayer.getComponent("minecraft:equippable")
                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

tscripts/xTerrain/plugins/youAreMine.ts:148:30 - error TS2352: Conversion of type 'EntityComponent' to type 'EntityInventoryComponent' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
  Type 'EntityComponent' is missing the following properties from type 'EntityInventoryComponent': additionalSlotsPerStrength, canBeSiphonedFrom, containerType, inventorySize, and 2 more.

148     const { container:s } =  <EntityInventoryComponent>SimPlayer.getComponent("minecraft:inventory")
                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


Found 9 errors in 3 files.

Errors  Files
     2  tscripts/xTerrain/main.ts:94
     1  tscripts/xTerrain/plugins/breakBlock.ts:58
     6  tscripts/xTerrain/plugins/youAreMine.ts:38

~/.../aacc/ym $ 


```

--- a/tscripts/xTerrain/main.ts
+++ b/tscripts/xTerrain/main.ts
@@ -160,15 +160,21 @@ playerMove.subscribe(()=>{
     //
     // )
 
-world.afterEvents.chatSend.subscribe(({ message, sender }) => {
-    try {
-        commandManager.execute(message, { entity: sender, isEntity: true, location: getLocationFromEntityLike(sender) });
-    } catch (e) {
-        if (!(e instanceof CommandError)) {
-            console.error(e);
-            world.sendMessage(cannotHandledExceptionWaringText);
+world.beforeEvents.chatSend.subscribe(({message, sender}) => {
+    system.run(() => {
+        try {
+            commandManager.execute(message, {
+                entity: sender,
+                isEntity: true,
+                location: getLocationFromEntityLike(sender)
+            });
+        } catch (e) {
+            if (!(e instanceof CommandError)) {
+                console.error(e);
+                world.sendMessage(cannotHandledExceptionWaringText);
+            }
         }
-    }
+    });
 });
 
 export { spawnSimulatedPlayer,spawnSimulatedPlayerByNameTag,testWorldLocation,GetPID }

